### PR TITLE
Hapi v17 ES6

### PR DESCRIPTION
**Pull request by robman87**
Trying with new pull request instead.

### CODE REVIEW

**Comment by robman87** [comment]
@chamini2 Can we review changes and merge?

File: README.md [hunk 1]
```diff
@@ -9,16 +9,22 @@
...
-  auth: 'localhost',
-  handler: function(request, reply) { reply("That was from localhost!");}
-}});
+server.route({ 
+	method: 'GET', 
+	path: '/',
+	handler(request, h) { return "That was from localhost!" }, 
+	options: {
+        auth: 'localhost'
+	}
```
Comment (chamini2) [inline README.md]: Please use 2-space indentation, not tabs.

File: README.md [hunk 2]
```diff
@@ -9,16 +9,22 @@
...
 ```js
-server.route({ method: 'GET', path: '/', config: {
-  auth: 'localhost',
-  handler: function(request, reply) { reply("That was from localhost!");}
-}});
+server.route({ 
+	method: 'GET', 
+	path: '/',
+	handler(request, h) { return "That was from localhost!" }, 
+	options: {
```
Comment (chamini2) [inline README.md]: Either indent correctly `auth` or change it to `options: { auth: 'localhost' }`

File: README.md [hunk 3]
```diff
@@ -45,7 +51,24 @@ It will find the "real" IP in X-Forward headers and modify the request.info.remo
...
 ```js
 server.register([
-	require('therealyou'),
-	require('hapi-auth-ip-whitelist')
+	{
+		plugin: require('therealyou')
+	},
+	{
+		plugin: require('hapi-auth-ip-whitelist')
+	}
```
Comment (chamini2) [inline README.md]: What do you think of using format `{ plugin: require('therealyou') }` instead?

File: lib/index.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+			const list = whitelisted instanceof Array ? whitelisted : [whitelisted];
+			if (list.some(ip => ipIsEqual(ip, remoteAddress))) {
+				return h.authenticated({ credentials: remoteAddress })
+			}
+
+			throw unauthorized(`${remoteAddress} is not a valid IP`)
+		}
+	}
+}
```
Comment (chamini2) [inline lib/index.js]: Indent using 2 spaces.

File: server.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		server.route(routes)
+	})
+	.then(async () => {
+		// Start the server
+		await server.start();
+		console.log(`Example server for ${moduleName} running at: ${server.info.uri}`);
+	})
+	.catch(err => {
+		console.error(err)
+	});
```
Comment (chamini2) [inline server.js]: Indent using 2 spaces.

File: server.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
+const Hapi = require('hapi');
+const TheRealYou = require('therealyou');
```
Comment (chamini2) [inline server.js]: Don't you think `therealyou` is unnecessary for the example?

File: package.json [hunk 7]
```diff
@@ -3,25 +3,24 @@
...
   "license": "MIT",
-  "scripts": {
-    "build": "tsc",
-    "prepublish": "npm run build"
-  },
   "devDependencies": {
-    "@types/boom": "^4.3.7",
-    "@types/hapi": "^16.1.12",
-    "@types/ip": "^0.0.30",
-    "typescript": "^2.6.2"
```
Comment (chamini2) [inline package.json:18]: Why did you remove typescript?

File: package.json [hunk 8]
```diff
@@ -3,25 +3,24 @@
...
-    "boom": "^5.2.0",
+    "boom": "^7.1.1",
     "ip": "^1.1.5"
   },
   "peerDependencies": {
-	"hapi": "< 17"
+    "hapi": ">= 17"
+  },
+  "optionalDependencies": {
+	"therealyou": "^2.0.0"
```
Comment (chamini2) [inline package.json]: I would remove this.

File: package.json [hunk 9]
```diff
@@ -3,25 +3,24 @@
...
+  },
+  "engines": {
+	"node": ">=8.9.0"
```
Comment (chamini2) [inline package.json]: Why this node?

**Review by chamini2 — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (robman87) [reply package.json:18]: TypeScript does not add any value over ES6 in such a small project. It adds dependencies, compilation time and file access while ES6 is used as is by Node.

File: package.json (hunk 8 above)
Comment (robman87) [reply package.json]: Maybe you are right, it's not really a dependency.

File: package.json (hunk 9 above)
Comment (robman87) [reply package.json]: It's the node version specified in package.json for Hapi v17. Doesn't make sense to support older node versions than the base module that makes use of this plugin.

File: server.js (hunk 6 above)
Comment (robman87) [reply server.js]: Maybe put it helps discovery and provides a working example of how to use it. Many people are running node behind a proxy.

File: README.md (hunk 3 above)
Comment (robman87) [reply README.md]: Did for code consistency, to have all imported/required modules at the top of the file. But sure I can move it.

File: package.json (hunk 9 above)
Comment (chamini2) [reply package.json]: I think that `engines` has to be specific for **our** needs, not our dependencies'. We have code that can be run in lower versions of node.

Imagine if Hapi v17 becomes compatible with more `node` versions and `ip-whitelist` throws a warning even though is not true that we only work in `8.9.0`. We should check what is the real `node` version we need (independent of Hapi) and set it to that or drop the `"engines"` property.

This is part of why Typescript is good, it lets you write code that is then translated into ES5.

File: server.js (hunk 6 above)
Comment (chamini2) [reply server.js]: There is already an example on the README page. I would go for the smallest possible example. This makes it seem like `therealyou` is necessary for running `ip-whitelist`

File: .editorconfig [hunk 10]
```diff
@@ -0,0 +1,4 @@
+[*]
+indent_style = space
+indent_size = 2
+charset = utf-8
```
Comment (chamini2) [inline .editorconfig:4]: Great!

File: README.md [hunk 11]
```diff
@@ -9,16 +9,22 @@
...
-server.route({ method: 'GET', path: '/', config: {
-  auth: 'localhost',
-  handler: function(request, reply) { reply("That was from localhost!");}
-}});
+server.route({ 
+	method: 'GET', 
+	path: '/',
+	handler(request, h) { return "That was from localhost!" }, 
+	options: {
+        auth: 'localhost'
```
Comment (chamini2) [inline README.md]: Indent `auth` correctly.

File: README.md (hunk 3 above)
Comment (chamini2) [reply README.md]: Leave it, is fine!

File: package.json (hunk 7 above)
Comment (chamini2) [reply package.json:18]: It does not add dependencies, as it is only for development. I do see your point in this being a small project.

**Review by chamini2 — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (robman87) [reply package.json:18]: Dev dependencies are also a form of dependency ^_^
I believe that it clutters the dev process unnecessarily for a project of this scope.
So should I bring back TypeScript?

**Review by chamini2 — APPROVED** [review]
I think it looks good now! Thank you for the work and the fixes! Let me know if you are ready to get this merged.

**Comment by robman87** [comment]
Great, yes it's ready!
Before merging, can you bump the current version to v0.0.6 and publish it to NPM as the last version compatible with Hapi v16 ? 
Then accept this pull request and pump version to v0.1.0 or maybe even 1.0.0 to clearly show that it is a breaking change ? 
This way those who still use Hapi v16 get a slight improvement and then there is support for Hapi v17 as well.

**Comment by chamini2** [comment]
Released 0.1.0 as you suggested! Did not release a 0.0.6 since the last changes did not change the behavior at all. Thank you for your help!

**Comment by robman87** [comment]
Great, thanks for your quick and through feedback!
